lib/pbench/server/api/__init__.py

```diff
@@ -92,15 +92,36 @@ def _corsify_actual_response(response):
     return response
 
 
-class GraphQL(Resource):
+class CorsOptionsRequest:
+    def __init__(self, options_exception_msg):
+        self.options_exception_msg = options_exception_msg
+
+    # From the client side, an OPTIONS request is initiated before every POST to request CORS.
+    # On the server side, the server must specify the response headers to grant CORS accordingly.
+
+    def options(self):
+        try:
+            response = _build_cors_preflight_response()
+        except Exception as e:
+            app.logger.exception(self.options_exception_msg + " Raised: %s", e.__class__)
+            abort(400, message=self.options_exception_msg)
+        response.status_code = 200
+        return response
+
+
+class GraphQL(Resource, CorsOptionsRequest):
+    def __init__(self):
+        super(GraphQL, self).__init__("Bad options request for the Graphql query")
+
     def post(self):
         global app
         graphQL = app.config_graphql.get("server")
         json_data = request.get_json()
 
         if not json_data:
-            app.logger.debug("Invalid query object")
-            abort(400, message="Invalid query object in request")
+            app.logger.error("GraphQL: Invalid json object \n "
```

Comment:
Do we really want to be logging with explicit newlines? This can make a text log file harder to skim and to process with tools. I don't recall seeing this pattern elsewhere yet, though I may not have noticed. And if you were dumping a JSON object, I could see some value (or at least inevitability), but it seems unnecessary for a URI.

Reply:
make sense, but flask request.url does contain the actual json query https://flask.palletsprojects.com/en/1.1.x/api/#incoming-request-data

Reply:
Well, you could log the `base_url` to keep it short; but according to that doc page, `uri` isn't JSON payload, just the full user URL including the base URI and any query string, so it shouldn't be that bad.

Reply:
This particular comment thread seems to have been resolved.